arbitrum-client/src/client/contract_interaction.rs

```diff
@@ -637,44 +628,22 @@ impl ArbitrumClient {
     // -----------
 
     /// Sends a transaction, awaiting its confirmation and returning the receipt
-    pub async fn send_tx(
+    pub async fn send_tx<C: CallDecoder>(
         &self,
-        tx: ContractCall<MiddlewareStack, impl Detokenize>,
+        tx: DarkpoolCallBuilder<'_, C>,
     ) -> Result<TransactionReceipt, ArbitrumClientError> {
-        #[cfg(feature = "tx-metrics")]
-        incr_inflight_txs();
-
-        // Set the gas price to 2x the latest basefee for simplicity
-        let latest_block = self
-            .client()
-            .get_block(BlockNumber::Latest)
-            .await
-            .map_err(err_str!(ArbitrumClientError::Rpc))?
-            .ok_or(ArbitrumClientError::Rpc("No latest block found".to_string()))?;
-
-        let latest_basefee = latest_block
-            .base_fee_per_gas
-            .ok_or(ArbitrumClientError::Rpc("No basefee found".to_string()))?;
-
-        let tx = tx.gas_price(latest_basefee * 2);
```

Comment:
what is the default gas price estimation like in `alloy` I remember in ethers, it's some fancy median of past prices (or something), lower-bounded by a fixed constant like 3 gwei or something which is clearly tailored to mainnet expectations.

if the logic is the same in alloy, we should continue using our custom estimation here